### PR TITLE
Feat: Adding a port-forward mode proxy entrypoint

### PR DESCRIPTION
**Pull request by yue9944882**

### CODE REVIEW

File: pkg/util/portforward.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+		return err
+	}
+	defer streamConn.Close()
+
+	// create error stream
+	headers := http.Header{}
+	headers.Set(v1.StreamType, v1.StreamTypeError)
+	headers.Set(v1.PortHeader, fmt.Sprintf("%d", r.targetPort))
+	headers.Set(v1.PortForwardRequestIDHeader, strconv.Itoa(r.reqId))
+	r.reqId++
```
Comment (qiujian16) [inline pkg/util/portforward.go:122]: add a todo here also? I think you would need to at least lock here.

File: pkg/util/portforward.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+		runtime.HandleError(fmt.Errorf("error creating forwarding stream for port %d -> %d: %v", r.targetPort, r.targetPort, err))
+		return err
+	}
+
+	localError := make(chan struct{})
+	remoteDone := make(chan struct{})
+
+	go func() {
+		// Copy from the remote side to the local port.
+		if _, err := io.Copy(conn, dataStream); err != nil && !strings.Contains(err.Error(), "use of closed network connection") {
```
Comment (qiujian16) [inline pkg/util/portforward.go:156]: why do we need to exclude the closed error?

File: pkg/util/portforward.go (hunk 1 above)
Comment (yue9944882) [reply pkg/util/portforward.go:126]: added a lock

File: pkg/util/portforward.go (hunk 2 above)
Comment (yue9944882) [reply pkg/util/portforward.go:162]: https://github.com/kubernetes/kubernetes/blob/cd6ffff85d257ff9067d59339f2ffdbcc66dc164/staging/src/k8s.io/client-go/tools/portforward/portforward.go#L309-L310

actually this line is copied from the original client-go's port-forward server. for now the client-go library cannot be reused by cluster-proxy due to some limitation

**Review by qiujian16 — COMMENTED** [review]
/lgtm

**Pull request merged**